models.js

```diff
@@ -14,27 +14,43 @@ const fetchAllTopics = (sort_by = "slug") => {
 
 const fetchArticleById = (article_id) => {
   return db
-    .query(
-      `SELECT 
-        author,
-        body,
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ msg: "Not Found" });
+      }
+      return rows[0];
+    });
+};
+
+const fetchAllArticles = (data) => {
+  let query = `SELECT 
+        author, 
         title,
         article_id,
         topic,
         created_at,
         votes,
         article_img_url,
-        (SELECT COUNT(*) FROM comments WHERE comments.article_id = articles.article_id) AS comment_count 
-        FROM articles 
-        WHERE article_id = $1`,
-      [article_id]
-    )
-    .then(({ rows }) => {
-      if (rows.length === 0) {
-        return Promise.reject({ msg: "Not Found" });
-      }
-      return rows[0];
+        (SELECT COUNT(*) FROM comments WHERE comments.article_id = articles.article_id) AS comment_count
+        FROM articles`;
+
+  if (data) {
+    query += ` WHERE topic = $1 ORDER BY created_at DESC`;
+    return db.query(query, [topic]).then((result) => {
+      return result.rows.map((article) => {
+        delete article.body;
+        return article;
+      });
```

Comment:
is this step necessary? We wouldn't get a body if it's not been included in the SELECT statement, so we can handle this without extra logic